src/ocaml/typing/408/typemod.ml

```diff
@@ -189,7 +188,7 @@ let initial_env ~loc ~safe_string ~initially_opened_module
           match after with
           | [] -> None
           | units :: after ->
-              if String.Set.mem m units then
+              if StringSet.mem m units then
```

Comment:
In https://github.com/ocaml/merlin/commit/828a79d6121913a61294753555526a7d1739797e (part of the 4.09 support) I propose to keep `String.Set` to stay close to the upstream typechecker style.

With this patch, `Std.String.Set` and `Std.String.Map` are the labelled versions (incompatible with the typechecker code) while `Misc.String.Set` and `Misc.String.Map` are the unlabelled ones.
It works as long as compiler modules open `Misc` module and Merlin modules open `Std` after `Misc`.

Reply:
828a79d is a good idea, I extracted it from your PR and pushed it directly on master.
I'll rebase this PR once more (and I guess you'll have to rebase yours)